test/extensions/extensions_build_system.bzl

```diff
@@ -1,54 +1,66 @@
 load("//bazel:envoy_build_system.bzl", "envoy_benchmark_test", "envoy_cc_benchmark_binary", "envoy_cc_mock", "envoy_cc_test", "envoy_cc_test_binary", "envoy_cc_test_library", "envoy_py_test")
 load("@envoy_build_config//:extensions_build_config.bzl", "EXTENSIONS")
 
-def _apply_to_extension_allow_for_test(func, name, extension_name, **kwargs):
+def _apply_to_extension_allow_for_test(func, name, extension_name, additional_extension_names, **kwargs):
     if not extension_name in EXTENSIONS:
         return
 
+    for extension_name in additional_extension_names:
+        if not extension_name in EXTENSIONS:
+            return
+
     func(name, **kwargs)
 
 # All extension tests should use this version of envoy_cc_test(). It allows compiling out
 # tests for extensions that the user does not wish to include in their build.
 # @param extension_name should match an extension listed in EXTENSIONS.
+# @param each item of additional_extension_names should match an extension listed in EXTENSIONS.
 def envoy_extension_cc_test(
         name,
         extension_name,
+        additional_extension_names = [],
```

Comment:
There's a common gotcha in python about argument default values being mutable and shared across calls.  The recommended pattern in python is to use `= None` instead of `= []` for list types and handle the case where the argument is set to None.

In this case the handling would be in `_apply_to_extension_allow_for_test` and would be something like:
```
if additional_extension_names:
    for extension_name in additional_extension_names:
        ...
```